tools/gce-google-keys-to-cmek/main.py

```diff
@@ -0,0 +1,207 @@
+    print('Deleting disk project={0}, zone={1}, disk={2}'.format(project, zone, disk))
+    operation = compute.disks().delete(project=project, zone=zone, disk=disk).execute()
+    result = wait_for_zonal_operation(compute, project, zone, operation)
+    print('Deleting disk project={0}, zone={1}, disk={2} complete.'.format(project, zone, disk))
+    return result
+
+def create_disk(compute, project, region, zone, snapshot_name, disk_name, disk_type, key_name):
+    """Creates a new CMEK encrypted persistent disk from a snapshot"""
+    sourceSnapshot = 'projects/{0}/global/snapshots/{1}'.format(project, snapshot_name)
+    body = {
+        'name': disk_name,
+        'sourceSnapshot': sourceSnapshot,
+        'type': disk_type,
+        'diskEncryptionKey': {
+            'kmsKeyName': key_name,
+        },
+    }
+    start = 'Creating new disk project={0}, zone={1}, name={2} sourceSnapshot={3}, kmsKeyName={4}'
+    finish = start + ' complete'
+    print(start.format(project, zone, disk_name, sourceSnapshot, key_name))
+    operation = compute.disks().insert(project=project, zone=zone,  body=body).execute()
+    result = wait_for_zonal_operation(compute, project, zone, operation)
+    print(finish.format(project, zone, disk_name, sourceSnapshot, key_name))
+    return result
+
+def start_instance(compute, project, zone, instance):
+    print('Starting project={0}, zone={1}, instance={2}'.format(project, zone, instance))
+    operation = compute.instances().start(project=project, zone=zone, instance=instance).execute()
+    result = wait_for_zonal_operation(compute, project, zone, operation)
+    print('Starting project={0}, zone={1}, instance={2} complete.'.format(project, zone, instance))
+    return result
+
+def stop_instance(compute, project, zone, instance):
+    print('Stopping project={0}, zone={1}, instance={2}'.format(project, zone, instance))
+    operation = compute.instances().stop(project=project, zone=zone, instance=instance).execute()
+    result = wait_for_zonal_operation(compute, project, zone, operation)
+    print('Stopping project={0}, zone={1}, instance={2} complete.'.format(project, zone, instance))
+    return result
+
+def wait_for_global_operation(compute, project, operation):
+    operation = operation['name']
+    def build():
+        return compute.globalOperations().get(
+            project=project,
+            operation=operation)
+    return _wait_for_operation(operation, build)
+
+def wait_for_zonal_operation(compute, project, zone, operation):
+    operation = operation['name']
+    def build():
+        return compute.zoneOperations().get(
+            project=project,
+            zone=zone,
+            operation=operation)
+    return _wait_for_operation(operation, build)
+
+def _wait_for_operation(operation, build_request):
+    """Helper for waiting for operation to complete."""
+    print('Waiting for {0}'.format(operation), end='')
+    while True:
+        print('.', end='')
```

Comment:
Consider refactoring to a spinner instead of printing an awkward number of periods w/o a newline.